### PR TITLE
[[ Bug 7303 ]] Implement autocomplete in script editor

### PR DESCRIPTION
**Pull request by montegoulding**
This patch adds autocomplete to the script editor. Autocomplete providers
are generated by object introspection using revAvailableHandlers and
revAvailableVariables. Autocomplete providers are also generated from
the syntax statements in the documentation database.

### CODE REVIEW

**Comment by montegoulding** [comment]
autocomplete
autocomplete

**Comment by montegoulding** [comment]
This PR now depends on https://github.com/livecode/livecode/pull/5739

File: Toolset/palettes/script editor/behaviors/revsecommoneditorbehavior.livecodescript [hunk 1]
```diff
@@ -3109,3 +3214,1204 @@ private command __LookForPair pChar, pPairChar, pStep, @xCharNum
...
+end __UpdateAutoCompleteList
+
+private function __IconForSymbolClass pSymbolClass
+   switch
+      case pSymbolClass contains "command"
+         return "<b>C</b>"
+      case pSymbolClass contains "function"
+         return "<b>F</b>"
+      case pSymbolClass contains "command"
+         return "<b>V</b>"
```
Comment (BerndN) [inline Toolset/palettes/script editor/behaviors/revsecommoneditorbehavior.livecodescript]: Is this needed? "command" is already handled above.

Comment (montegoulding) [reply Toolset/palettes/script editor/behaviors/revsecommoneditorbehavior.livecodescript]: Thanks @BerndN nice catch. Just a copy, paste error. BTW I'm looking for ideas on better chars for comment. I originally had `#` but then it looked odd in the list having `# -- comment`. So I went with something different `~`... But I'm open to ides.

**Pull request closed without merging**